api/cluster/src/main/java/org/pentaho/big/data/api/cluster/NamedCluster.java

```diff
@@ -35,6 +35,10 @@
 
   void setName( String name );
 
+  String getShimIdentifier();
+
+  void setShimIdentifier( String shimIdentifier );
+
```

Comment:
That was done according to previous fix version. There are a lot of places currently where we has logic depending on shim identifier from named cluster. We just had TODO comments there. 
like: https://github.com/pentaho/pentaho-yarn-kettle-plugin/pull/257/commits/21b417ce56985481f8af0f72d96aba2c88e02d7f#diff-aa3749266b9ba77985e122dd2ae9998fR94
or: https://github.com/pentaho/big-data-plugin/pull/795/commits/fa2539f4a8fcd13dc6c363304eabdfff24c2da9f#diff-f4094c7836216afd4e4b55ee0bddd2a8R55 
NamedCluster model should have such property.